client/lock.go

```diff
@@ -238,33 +239,48 @@ func (c *lockClient) waitDelete(pfx string, myRev int64) {
 			return
 		}
 
-		watcher, _ := c.watchClient.Watch()
-		err = watcher.Send(&xlineapi.WatchRequest{
-			RequestUnion: &xlineapi.WatchRequest_CreateRequest{
-				CreateRequest: &xlineapi.WatchCreateRequest{
-					Key: lastKey,
-				},
-			},
-		})
-		if err != nil {
-			logger.Warn("Watch lease ID fail.", zap.Error(err))
-		}
-		for {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		watcher, _ := c.watchClient.Watch(ctx, lastKey)
+		// err = watcher.Send(&xlineapi.WatchRequest{
+		// 	RequestUnion: &xlineapi.WatchRequest_CreateRequest{
+		// 		CreateRequest: &xlineapi.WatchCreateRequest{
+		// 			Key: lastKey,
+		// 		},
+		// 	},
+		// })
+		// if err != nil {
+		// 	logger.Warn("Watch lease ID fail.", zap.Error(err))
+		// }
+		for r := range watcher {
 			f := false
-			watchRes, err := watcher.Recv()
-			if err != nil {
-				logger.Warn("Watch lease ID fail.", zap.Error(err))
-			}
-			for _, e := range watchRes.Events {
+			for _, e := range r.Events {
 				if e.Type == xlineapi.Event_DELETE {
 					f = true
```

Comment:
Will the assignment to f will be reordered?

Reply:
This is not a problem to be solved in this pr, the change is just to make the ci work, we can discuss that in #19